### ISSUE TITLE
Icon URL is missing from tabs sync data

### ISSUE THREAD

**Issue by lougeniaC64** [mozilla-mobile/firefox-ios#11851]
## Steps to reproduce
1. Open Firefox iOS
1. Sign in
1. Open a few tabs
1. Click Sync Now from the settings menu
1. Navigate to `about:sync` in desktop
  * This will require installing the [About Sync add-on](https://addons.mozilla.org/en-US/firefox/addon/about-sync)
1. Scroll to the bottom of the page and expand the tabs items
  * You may need to trigger a sync in desktop before the tabs created in step 3 appear

### Expected behavior
Tabs records for iOS clients should have populated icon values.

### Actual behavior
Tabs records have null icon values.

### Notes
This is a longstanding issue that may be related to issues with the population of other tabs properties like [lastUsed](https://github.com/mozilla-mobile/firefox-ios/issues/11769) and [title](https://github.com/mozilla-mobile/firefox-ios/issues/11742). I am seeing that nil values are passed into the icon property in [Tab.toRemoteTab](https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/Browser/Tab.swift#L402), but when I change `icon: nil` to `icon: tab.currentFaviconUrl ?? nil` I'm still not seeing the icon value come through.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4898)

**Comment by OrlaM** [comment]
@lougeniaC64 What is the purpose of this field? I didn't go through the entire history but it looks like this hasn't been populated for years, or potentially ever. Is it's absence causing bugs on other platforms that are sync'd with iOS devices or do they just ignore this and retrieve the favicon directly?

Some context for my question. We are in the process of refactoring how favicons work and we will be starting to refactor tabs management very soon. In an ideal world I'd like to remove as much of the favicon logic away from tabs as I can. Supporting this use case isn't a massive effort but it is complexity I'd like to avoid if it has little impact.

**Comment by lougeniaC64** [comment]
@OrlaM I believe the `icon` property is where the favicon URL should be stored. If you look at the tab records in `about:sync` you'll see the records from desktop have the `icon` property populated. This is one of my desktop tab records for reference:


![Screen Shot 2023-01-03 at 5 51 00 PM](https://user-images.githubusercontent.com/5533446/210454497-d2ed897f-7cbe-48c0-9afa-d5957361fb37.png)

This bug appears to be long-standing and was discovered during the firefox view dogfooding.

**Comment by OrlaM** [comment]
What is the product impact of this value not being populated? Does it cause some bugs or issues that are visible to an end user?

**Comment by lougeniaC64** [comment]
I believe the highest visibility this issue had was during the Firefox View testing and I think even then it wasn't considered a high priority. Even now as I look at the Tab pickup section of my Firefox View I can see some tabs coming from my iPhone that don't have favicons (there's a default globe image displayed instead).

All that being said I imagine that the icon property not being populated isn't a high priority or severity issue.

**Comment by OrlaM** [comment]
Ok sounds good. I will incorporate this fix into the refactor work.

**Comment by nishant2718** [comment]
Although the faviconURL value on tab is more reliable now after recent changes, the syncs don't seem to happen at the timing we expect. 

What I'd want to see is:
1. Page metadata is populated first (MetaDataParserHelper)
2. RemoteTab creation / storage occurs (TabManager)
3. Finally, a sync occurs via the observed tab events in AccountSyncHandler

Instead, what I see is the above steps happening in some random order each time (and the set of tabs is looped over twice). 

So basically, syncs happen at times when that property isn't available yet. :/ A proper fix ... might be with the Tabs rework?
